From 0bd969df5e9f83c59dd21397f347701088235459 Mon Sep 17 00:00:00 2001
From: "Tan, Raymond" <raymond.tan@intel.com>
Date: Tue, 5 Nov 2019 14:07:39 +0800
Subject: [PATCH] dma: dw: Disable support for Elkhart Lake PSE DMA

This patch remove the ACPI IDs and PCI DIDs for Elkhart Lake PSE DMA,
thus removing the support for this release. The functionality is
still being enabled and debugged.

Signed-off-by: Tan, Raymond  <raymond.tan@intel.com>
---
 drivers/dma/dw/pci.c      | 5 -----
 drivers/dma/dw/platform.c | 6 ------
 2 files changed, 11 deletions(-)

diff --git a/drivers/dma/dw/pci.c b/drivers/dma/dw/pci.c
index 051a4df..8f5f1a5 100644
--- a/drivers/dma/dw/pci.c
+++ b/drivers/dma/dw/pci.c
@@ -115,11 +115,6 @@ static const struct pci_device_id dw_pci_id_table[] = {
 	{ PCI_VDEVICE(INTEL, 0x2286), (kernel_ulong_t)&dw_dma_chip_pdata },
 	{ PCI_VDEVICE(INTEL, 0x22c0), (kernel_ulong_t)&dw_dma_chip_pdata },
 
-	/* Elkhart Lake iDMA 32-bit (PSE DMA) */
-	{ PCI_VDEVICE(INTEL, 0x4bb4), (kernel_ulong_t)&idma32_xbar_pdata },
-	{ PCI_VDEVICE(INTEL, 0x4bb5), (kernel_ulong_t)&idma32_xbar_pdata },
-	{ PCI_VDEVICE(INTEL, 0x4bb6), (kernel_ulong_t)&idma32_xbar_pdata },
-
 	/* Haswell */
 	{ PCI_VDEVICE(INTEL, 0x9c60), (kernel_ulong_t)&dw_dma_chip_pdata },
 
diff --git a/drivers/dma/dw/platform.c b/drivers/dma/dw/platform.c
index bffc79a..e0766d4 100644
--- a/drivers/dma/dw/platform.c
+++ b/drivers/dma/dw/platform.c
@@ -147,12 +147,6 @@ static const struct acpi_device_id dw_dma_acpi_id_table[] = {
 	{ "INTL9C60", (kernel_ulong_t)&dw_dma_chip_pdata },
 	{ "80862286", (kernel_ulong_t)&dw_dma_chip_pdata },
 	{ "808622C0", (kernel_ulong_t)&dw_dma_chip_pdata },
-
-	/* Elkhart Lake iDMA 32-bit (PSE DMA) */
-	{ "80864BB4", (kernel_ulong_t)&idma32_xbar_pdata },
-	{ "80864BB5", (kernel_ulong_t)&idma32_xbar_pdata },
-	{ "80864BB6", (kernel_ulong_t)&idma32_xbar_pdata },
-
 	{ }
 };
 MODULE_DEVICE_TABLE(acpi, dw_dma_acpi_id_table);
-- 
2.7.4

